File: iosApp/iosApp/Components/GhostButton.swift
import SwiftUI

/// Low-emphasis outlined action button — mirrors web's `GhostButton` (design-system
/// "Secondary / Outlined" style). Used for supporting actions like "Join this Read" /
/// "Opt out" or "Update" progress.
struct GhostButton: View {
    let text: String
    let onClick: () -> Void
    var isEnabled: Bool = true

    var body: some View {
        Button(action: onClick) {
            Text(text)
                .font(.kluvsButtonSecondary)
                .foregroundColor(.primary)
                .padding(.vertical, 8)
                .padding(.horizontal, 12)
                .overlay(
                    RoundedRectangle(cornerRadius: 12)
                        .strokeBorder(Color(uiColor: .separator), lineWidth: 1)
                )
        }
        .disabled(!isEnabled)
        .opacity(isEnabled ? 1 : 0.5)
    }
}

#Preview {
    GhostButton(text: "Join this Read", onClick: {})
        .padding()
}

File: composeApp/src/androidMain/kotlin/com/ivangarzab/kluvs/ui/components/GhostButton.kt
package com.ivangarzab.kluvs.ui.components

import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.PreviewLightDark
import androidx.compose.ui.unit.dp
import com.ivangarzab.kluvs.designsystem.theme.KluvsTheme

/**
 * Low-emphasis outlined action button — mirrors web's `GhostButton` (design-system
 * "Secondary / Outlined" style, see design-system/docs/buttons.md). Used for
 * supporting actions like "Join this Read" / "Opt out" or "Update" progress.
 */
@Composable
fun GhostButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
) {
    OutlinedButton(
        onClick = onClick,
        modifier = modifier,
        enabled = enabled,
        shape = RoundedCornerShape(12.dp),
        contentPadding = PaddingValues(vertical = 8.dp, horizontal = 8.dp)
    ) {
        Text(
            text = text,
            color = MaterialTheme.colorScheme.onBackground,
            style = MaterialTheme.typography.labelMedium
        )
    }
}

@PreviewLightDark
@Composable
fun Preview_GhostButton() = KluvsTheme {
    GhostButton(text = "Join this Read", onClick = {})
}
